src/create_shards.c

```diff
@@ -422,16 +428,43 @@ ParseWorkerNodeFile(char *workerNodeFilename)
 			continue;
 		}
 
-		/* parse out the node name and node port */
-		parsedValues = sscanf(workerNodeLine, workerLinePattern, nodeName, &nodePort);
-		if (parsedValues != 2)
+		/* parse line; node name is required, but port is optional */
+		fieldCount = sscanf(workerNodeLine, workerLinePattern, nodeName, nodePortString);
+
+		/* adjust field count for zero based indexes */
+		fieldCount--;
+
+		/* raise error if no fields were assigned */
+		if (fieldCount < WorkerNameIndex)
 		{
 			ereport(ERROR, (errcode(ERRCODE_CONFIG_FILE_ERROR),
 							errmsg("could not parse worker node line: %s",
 								   workerNodeLine),
 							errhint("Lines in the worker node file consist of a node "
-									"name and port separated by whitespace. Lines that "
-									"start with a '#' character are skipped.")));
+									"name and optional port separated by whitespace. "
```

Comment:
> * Can't we use a boolean variable to indicate whether there is an error or not? Then, use that variable and write the `ereport` only once?

Hm… I had considered a `goto` and decided that was too unconventional, especially since the repetition here is only once (if I had three copies of the error I'd definitely do something to cut back on repetition). But I think the boolean could work…

> * If that is not possible, its better that we make the layout exactly the same. Second `ereport`'s `errhint` is splitted into 4 lines, whereas the first one is splitted into 3 lines.

Eh, that's more because I hit the 90 character limit, but I do remember thinking it was unsightly. I'll try out the boolean.